### PR TITLE
#3159 Added new get started guide

### PR DESCRIPTION
**Pull request by vikram-dagger**
Relates to https://github.com/dagger/dagger/issues/3159

Note: This draft will change as the API changes.

### CODE REVIEW

**Comment by samalba** [comment]
@vikram-dagger I think the markdown error is valid, here is how it renders locally:

Screen Shot 2022-09-28 at 3 53 38 PM

And on the netlify deploy (production build):

Screen Shot 2022-09-28 at 3 57 09 PM

I assume you're looking for the latter, which means removing the indentation.

**Comment by mircubed** [comment]
@jpadams right now, Vikram has a placeholder for extensions, but we were discussing that it needs to be linked to the universe somehow. Should he keep that section as a placeholder or should we assume that the extension section will point to the universe experience?

**Comment by vikram-dagger** [comment]
> @jpadams right now, Vikram has a placeholder for extensions, but we were discussing that it needs to be linked to the universe somehow. Should he keep that section as a placeholder or should we assume that the extension section will point to the universe experience?

My plan here would be to generate the extension pages and their contents automatically as follows:
- have a machine-readable file (which could come from the universe system) containing the list of extensions and their source repos
- each source repo would include an example file (and maybe others) in a standard form
- the docs system would have a script/plugin that would read this information and produce one page for each extension with the relevant content

Open to further ideas on this. Ideally, we don't want to custom-craft these extension pages but rather have them auto-generated, similar to the API reference.

**Comment by vikram-dagger** [comment]
> @vikram-dagger I think the markdown error is valid, here is how it renders locally:

I don't see this in my local render. It looks like the netlify version when I render locally. Not sure if you were looking at an older commit (I had a similar error before but I fixed it).

> And on the netlify deploy (production build):
> 
> I assume you're looking for the latter, which means removing the indentation.

When I remove indentation, the linter returns a different error

```
docs/get-started/bvtz9-get-started.md:126:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
docs/get-started/bvtz9-get-started.md:127:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
docs/get-started/bvtz9-get-started.md:128:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
```

**Comment by vikram-dagger** [comment]
> @vikram-dagger I think the markdown error is valid, here is how it renders locally:

I don't see this in my local render. It looks like the netlify version when I render locally. Not sure if you were looking at an older commit (I had a similar error before but I fixed it).

> And on the netlify deploy (production build):
> 
> I assume you're looking for the latter, which means removing the indentation.

@samalba When I remove indentation for the list, the linter passes but the list numbering gets reset so that step 4 of the section is renumbered as 1. The indentation of the list items is also not correct in the render, as they should be indented under the ``Let's break this query...` line.

image

IMO The correct approach is to have the items under `Let's break this query...` indented 2 levels as they are a sub-list within the main list. When done in this way, the sub-list renders correctly and the numbering for step 4 is also retained. 

image

However the linter throws the error `docs/get-started/bvtz9-get-started.md:126 MD046/code-block-style Code block style [Expected: fenced; Actual: indented]`
 
Open to any ideas to correct this.

File: docs/get-started/bvtz9-get-started.md [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+
+1. In a new terminal, start the Dagger API server using the following command.
+
+  ```bash
+  cloak dev
+  ```
+
+  By default, this will start the Dagger API server on the local host at port 8080. Starting the server will also start an interactive GraphQL query console on the same port.
+
+1. Access the query console by browsing to `http://localhost:8080/`. Here's what you will see.
```
Comment (slumbering) [inline docs/get-started/bvtz9-get-started.md:100]: Hey @vikram-dagger ! Should it be `2.` and line 106 `3.` etc.

Comment (vikram-dagger) [reply docs/get-started/bvtz9-get-started.md:100]: Not necessary, it will auto-number. See https://gist.github.com/PeeHaa/3489721

**Comment by vikram-dagger** [comment]
Closing this as we plan to start with a guide based on Go initially.

**Pull request closed without merging**